append an abac rule for $KUBE_USER.

cjcullen opened the pull request:
Allows the specified basic-auth credentials to be authorized against all resources.

Fixes #28869.

cjcullen committed:
append an abac rule for $KUBE_USER.

cjcullen commented:
Doesn't look like this is correct on our debian-based scripts.

adityakali commented:
I am not too familiar with salt, but the way I have seen this done in the past is to add a configurable variable in salt config, it your case, you can in https://github.com/kubernetes/kubernetes/blob/master/cluster/saltbase/salt/kube-apiserver/abac-authz-policy.jsonl something like:

{"apiVersion": "abac.authorization.kubernetes.io/v1beta1", "kind": "Policy", "spec": {"user":"{{kube_user}}", "namespace": "*", "resource": "*", "apiGroup": "*", "nonResourcePath": "*"}

and then
on debian: use saltstack to set the variable 
on gci: use 'sed' to substitute it with value of KUBE_USER

cjcullen commented:
@adityakali Yeah you're right. I'll do this in the more "salty" way.

cjcullen commented:
Okay, should be in a cleaner state now.

  - Added a line to abac-authz-policy.jsonl w/ the user specified by jinja variable `{{kube_user}}`
  - On Debian, configure-vm.sh puts `$KUBE_USER` into `grains.kube_user` for salt. Then the abac file is treated as a jinja template, so the value of `$KUBE_USER` gets into that line I added.
  - On GCI/Trusty, configure-helper.sh removes the jinja line, and replaces `{{kube_user}}` w/ the value of `$KUBE_USER` before copying the abac file.

PTAL

adityakali commented:
LGTM

zmerlynn commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -637,7 +637,12 @@ function start-kube-apiserver {
+  local -r abac_policy_json="${src_dir}/abac-authz-policy.jsonl"
+  remove-salt-config-comments "${abac_policy_json}"
+  sed -i -e "s@{{kube_user}}@${KUBE_USER}@g" "${abac_policy_json}"
+  cp ${abac_policy_json} /etc/srv/kubernetes/
```
nit: (and below) quotes: `"${abac_policy_json}"`

cjcullen replied:
Done and done.

zmerlynn commented:
LGTM except nit. Fix and feel free to self-apply label.

cjcullen commented:
Change was just a rebase to get past the docs verification error from yesterday. Reapplying LGTM.
